compiler/dyno/include/chpl/queries/Context.h

```diff
@@ -102,9 +103,27 @@ class Context {
   bool breakSet = false;
   size_t breakOnHash = 0;
   int numQueriesRunThisRevision_ = 0;
+  // tracks the nesting of queries, displayed during query tracing
+  int queryTraceDepth = 0;
+  // list of query names to ignore when tracing
+  const std::vector<std::string> queryTraceIgnoreQueries = {"idToTagQuery", "idToParentId"};
+  // list of colors to use for open/close braces depending on query depth
+  const std::vector<TermColorName> queryDepthColor  = {blue, bright_yellow, magenta};
 
   owned<std::ostream> queryTimingTraceOutput = nullptr;
 
+  // return an ANSI color code for this query depth, if supported by terminal
+  void setQueryDepthColor(int depth, std::ostream& os) {
+    auto color = queryDepthColor[depth % queryDepthColor.size()];
+    setTerminalColor(color, os);
+  }
+
+  void setTerminalColor(TermColorName colorName, std::ostream& os) {
+    if (terminalSupportsColor(getenv("TERM"))) {
```

Comment:
Why call getenv here vs. having e.g. currentTerminalSupportsColor? Also shouldn't we have something like this code:



```
      // Check if errors will be output to a tty. If not,
      // the format codes will just store "" and have no effect.
      if (isatty(fileno(stderr)) == 0) {
        isColorTerm = false;
      }
```


? Seems like we need that to be in a helper function at least and maybe combined into terminalSupportsColor

Reply:
good point, I refactored this into an instance variable so we don't have to keep checking with `getenv`. The only thing that isn't clear to me is if it's okay to disable color if _either_ `stdout` or `stderr` is a `tty`